### ISSUE TITLE
Failed to recognize a function

### ISSUE THREAD

**Issue by lvc** [universal-ctags/ctags#950]
Hello,

The latest version (9320ecff8751bfbed8daed9fc96124fea96d21a4) of the universal ctags failed to recognize SetImageChannelDepth symbol in the ImageMagick library header file (include/ImageMagick-6/magick/attribute.h, v6.9.4-4):

    extern MagickExport MagickBooleanType
      IdentifyImageMonochrome(const Image *,ExceptionInfo *),
      IsGrayImage(const Image *,ExceptionInfo *),
      IsMonochromeImage(const Image *,ExceptionInfo *),
      IsOpaqueImage(const Image *,ExceptionInfo *),
      SetImageChannelDepth(Image *,const ChannelType,const size_t),
      SetImageDepth(Image *,const size_t),
      SetImageType(Image *,const ImageType);

I use command:

    ctags -x --c-kinds=fpvx include/ImageMagick-6/magick/attribute.h | grep SetImageChannelDepth

The ctags lists only last symbol in this sentence (SetImageType). IdentifyImageMonochrome, IsGrayImage, IsMonochromeImage, IsOpaqueImage, SetImageChannelDepth and SetImageDepth are missed in the output.

Please help me to fix this issue. It's very important for the http://abi-laboratory.pro/tracker/ project.

Thank you.

**Comment by masatake** [comment]
Thank you for reporting.
Strange. Look at #945. At least `MagickExport` must be listed in command line with `-I` option.
The new C parser may have a bug about `-I` option handling. @pragmaware in our team is the author of parser. He is in vacation. So wait for awhile about `-I` option issue.

With `--languages=+OldC --language-force=OldC`, you can run old C parser:
```
$ u-ctags --verbose -I MagickExport --languages=+OldC --language-force=OldC --c-kinds='*' -o -  /tmp/foo.h
```
However, I can get nothing.

In other hand exuberant ctags works well:
```
$ e-ctags -I MagickExport --c-kinds='p' -o -  /tmp/foo.h
IdentifyImageMonochrome	/tmp/foo.h	/^  IdentifyImageMonochrome(const Image *,ExceptionInfo *),$/;"	p
IsGrayImage	/tmp/foo.h	/^  IsGrayImage(const Image *,ExceptionInfo *),$/;"	p
IsMonochromeImage	/tmp/foo.h	/^  IsMonochromeImage(const Image *,ExceptionInfo *),$/;"	p
IsOpaqueImage	/tmp/foo.h	/^  IsOpaqueImage(const Image *,ExceptionInfo *),$/;"	p
SetImageChannelDepth	/tmp/foo.h	/^  SetImageChannelDepth(Image *,const ChannelType,const size_t),$/;"	p
SetImageDepth	/tmp/foo.h	/^  SetImageDepth(Image *,const size_t),$/;"	p
SetImageType	/tmp/foo.h	/^  SetImageType(Image *,const ImageType);$/;"	p
```

I wonder why OldC parser cannot capture these names.

Basically I would like you to use the new C parser. So please, wait the response from @pragmaware.
I will inspect the old C parser.

**Comment by masatake** [comment]
The test input is simplified.

```
int f(void), g(void);
```

**Comment by lvc** [comment]
I will wait for a fix.

Thank you.

**Comment by lvc** [comment]
The commit fixes the issue with SetImageChannelDepth. But now ctags can't find similar MagickGetImageMean function in the `./include/ImageMagick-7/MagickWand/magick-image.h` header file of the imagemagick-7.0.1-6:

    ctags -x --c-kinds=fpvx -I WandExport ./include/ImageMagick-7/MagickWand/magick-image.h | grep MagickGetImageMean

The declaration of a set of functions is too long in this case:

    extern WandExport MagickBooleanType
    MagickAdaptiveBlurImage(MagickWand *,const double,const double),
    MagickAdaptiveResizeImage(MagickWand *,const size_t,const size_t),
    MagickAdaptiveSharpenImage(MagickWand *,const double,const double),
    MagickAdaptiveThresholdImage(MagickWand *,const size_t,const size_t,
    const double),
    MagickAddImage(MagickWand *,const MagickWand *),
    
    ... (skip about a hundred of functions in the same declaration)
    
    MagickGetImageMean(MagickWand *,double *,double *),
    
    ... (skip about a two hundreds of functions in the same declaration)
    
    MagickWriteImages(MagickWand *,const char *,const MagickBooleanType),
    MagickWriteImagesFile(MagickWand *,FILE *);
    
You can find the source of the header file here: http://git.imagemagick.org/repos/ImageMagick/blob/be80460638d2432ed6f7bdc93be1b342e30f6e9c/MagickWand/magick-image.h or inside the package: http://ftp.vim.org/ftp/pub/ImageMagick/ImageMagick-7.0.1-6.tar.xz

Thank you.

**Comment by masatake** [comment]
I simplified the reported input.


ctags captures nothing for input:
```
void
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void);
```

However, removing from one `f(void)` from the above input, ctags captures all `f` s.
```
void
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void), f(void),
f(void), f(void), f(void), f(void), f(void);
```

My trials are done with ctags with commits of #961.
```

**Comment by masatake** [comment]
Spotted.

```
	// More than 256 tokens is something like a big table, we
	// don't need it (for now).
	if(g_cxx.pTokenChain->iCount > 256)
		cxxTokenChainDestroyFirst(g_cxx.pTokenChain);

```

**Comment by pragmaware** [comment]
Yep, that's the problem. Removed the token limit.
Will merge the pull as soon as the checks pass.

**Comment by lvc** [comment]
It works!

Thanks a lot!
